File: orders-service/src/main/java/org/example/ordersservice/models/DetallePedido.java
package org.example.ordersservice.models;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;

@Entity
@Table(name = "detalle_pedido")
@Data
public class DetallePedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal precio;

    @Column(nullable = false)
    private Integer cantidad;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pedido_id")
    private Pedido pedido;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "producto_id")
    private Producto producto;
}

File: front/src/app/types/detalle-pedido.ts
export interface DetallePedido {
    id?: number;
    precio: number;
    cantidad: number;
    pedidoId?: number;
    productoId?: number;
}

export interface DetallePedidoInputDto {
    pedidoId: number;
    productoId: number;
    cantidad: number;
    precioUnitario: number;
}

export interface DetallePedidoOutputDto {
    id: number;
    productoId: number;
    nombreProducto: string;
    cantidad: number;
    precioUnitario: number;
    subtotal: number;
}
